boefjes/boefjes/plugins/kat_security_txt_downloader/main.py

```diff
@@ -0,0 +1,64 @@
+import ipaddress
+import json
+from os import getenv
+from typing import List, Tuple, Union
+
+import requests
+from forcediphttpsadapter.adapters import ForcedIPHTTPSAdapter
+from requests import Session
+
+from boefjes.job_models import BoefjeMeta
+
+
+def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
+    input_ = boefje_meta.arguments["input"]
+    netloc = input_["hostname"]["name"]
+    scheme = input_["ip_service"]["service"]["name"]
+    ip = input_["ip_service"]["ip_port"]["address"]["address"]
+
+    useragent = getenv("USERAGENT", default="OpenKAT")
+    session = requests.Session()
+
+    results = {}
+
+    for path in [".well-known/security.txt", "security.txt"]:
+        uri = f"{scheme}://{netloc}/{path}"
+
+        if scheme == "https":
+            session.mount(uri, ForcedIPHTTPSAdapter(dest_ip=ip))
+        else:
+            addr = ipaddress.ip_address(ip)
+            if addr.version == 6:
+                # IPv6 addresses need to be wrapped in brackets
+                netloc = f"[{ip}]"
+
+            uri = f"{scheme}://{netloc}/{path}"
+
+        response = do_request(netloc, session, uri, useragent)
```

Comment:
For http with IPv6 we would set Host header field to the IPv6 address because netloc is changed instead of the hostname. For IPv4 we don't seem to enforce the IP address at all?

Reply:
You are right. I overlooked ipv4. I'm following this example.